Core/test/qa-functional/src/org/sleuthkit/autopsy/modules/encryptiondetection/EncryptionDetectionTest.java

```diff
@@ -0,0 +1,198 @@
+import org.sleuthkit.autopsy.ingest.IngestJobSettings;
+import org.sleuthkit.autopsy.ingest.IngestJobSettings.IngestType;
+import org.sleuthkit.autopsy.ingest.IngestModuleError;
+import org.sleuthkit.autopsy.ingest.IngestModuleFactory;
+import org.sleuthkit.autopsy.ingest.IngestModuleIngestJobSettings;
+import org.sleuthkit.autopsy.ingest.IngestModuleTemplate;
+import org.sleuthkit.autopsy.testutils.DataSourceProcessorRunner;
+import org.sleuthkit.autopsy.testutils.DataSourceProcessorRunner.ProcessorCallback;
+import org.sleuthkit.autopsy.testutils.IngestJobRunner;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskData;
+
+public class EncryptionDetectionTest extends NbTestCase {
+
+    private static final String CASE_NAME = "EncryptionDetectionTest";
+    private static final Path CASE_DIRECTORY_PATH = Paths.get(System.getProperty("java.io.tmpdir"), CASE_NAME);
+    private static final File CASE_DIR = new File(CASE_DIRECTORY_PATH.toString());
+    private final Path IMAGE_PATH = Paths.get(this.getDataDir().toString(), "password_detection_test.img");
+
+    public static Test suite() {
+        NbModuleSuite.Configuration conf = NbModuleSuite.createConfiguration(EncryptionDetectionTest.class).
+                clusters(".*").
+                enableModules(".*");
+        return conf.suite();
+    }
+
+    public EncryptionDetectionTest(String name) {
+        super(name);
+    }
+
+    @Override
+    public void setUp() {
+        // Delete the test directory, if it exists
+        if (CASE_DIRECTORY_PATH.toFile().exists()) {
+            try {
+                FileUtils.deleteDirectory(CASE_DIRECTORY_PATH.toFile());
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
+                Assert.fail(ex);
+            }
+        }
+        assertFalse("Unable to delete existing test directory", CASE_DIRECTORY_PATH.toFile().exists());
+
+        // Create the test directory
+        CASE_DIRECTORY_PATH.toFile().mkdirs();
+        assertTrue("Unable to create test directory", CASE_DIRECTORY_PATH.toFile().exists());
+
+        try {
+            Case.createAsCurrentCase(Case.CaseType.SINGLE_USER_CASE, CASE_DIRECTORY_PATH.toString(), new CaseDetails(CASE_NAME));
+        } catch (CaseActionException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        }
+        assertTrue(CASE_DIR.exists());
+        ImageDSProcessor dataSourceProcessor = new ImageDSProcessor();
+        try {
+            ProcessorCallback callBack = DataSourceProcessorRunner.runDataSourceProcessor(dataSourceProcessor, IMAGE_PATH);
+            List<Content> dataSourceContent = callBack.getDataSourceContent();
+            assertEquals(1, dataSourceContent.size());
```

Comment:
We should use the assertEquals which includes a message argument so we can log a failure reason instead of just the exceptions stack trace.